src/components/display/components/LinkComponent.tsx

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react'
+import { useNavigation } from '@react-navigation/core'
+import { LinkArgs } from 'display-api'
+import { Linking } from 'react-native'
+import { TouchableRipple } from 'react-native-paper'
+import { redirectTo } from '../../../utils/helper'
+import { TextComponent } from './TextComponent'
+
+export const LinkComponent = ({ text, external, app_path, url }: LinkArgs) => {
+	const navigation = useNavigation()
+
+	const onPress = () => {
+		if (app_path) {
+			redirectTo(app_path, navigation)
```

Comment:
We will need to standardize `u/xx` for profile links. For web, accessing a user's page would use `u/louis`: their username, not their id. I think you'll need to change the mobile app to accept the name as the parameter for the profile root.

Reply:
Sure. In that case, we should update it when it's updated on the API or else the app features won't work. Do we have an existing API that takes username as parameter to get the block/user data?

Reply:
Yes, there is: `userByName`
It's the same as `userById` but queries the username.